packages/party_walls/tests/test_party_walls.py

```diff
@@ -1,89 +1,82 @@
+    return pd.read_csv(output_data_dir / "party_walls_nl.csv")
 
-@pytest.fixture(scope="function")
-def resource_db_connection():
-    """db_connection resource"""
-    return db_connection.configured({
-        "host": "localhost",
-        "port": 5432,
-        "user": os.environ.get("PYTEST_DB_USER"),
-        "password": os.environ.get("PYTEST_DB_PW"),
-        "dbname": "baseregisters_test"
-    })
 
+@asset(name="distribution_tiles_files_index")
+def mock_distribution_tiles_files_index(output_data_dir)  -> TilesFilesIndex:
+    return pickle.load(open(output_data_dir / "distribution_tiles_files_index.pkl", "rb"))
 
-@pytest.fixture(scope="session")
-def resource_file_store(output_data_dir):
-    """file_store resource pointing to the test output location"""
-    return file_store.configured({"data_dir": str(output_data_dir), })
 
+@asset(name="features_file_index")
+def mock_features_file_index(output_data_dir)  -> dict[str, Path]:
+    return pickle.load(open(output_data_dir / "features_file_index.pkl", "rb"))
 
-def test_distribution_tiles_files_index(export_dir_uncompressed):
-    """Can we parse the CityJSON tiles and return valid data?
-    Currently hardcoded for Balázs' local test data.
-    """
-    tile_ids = ('10/564/624', '10/564/626', '10/566/624', '10/566/626', '9/560/624')
 
+def test_distribution_tiles_files_index(context):
+    """Can we parse the CityJSON tiles and return valid data? """
+    
     result = distribution_tiles_files_index(
-        context=build_asset_context(
-            resources={"file_store": file_store.configured(
-                {"data_dir": str(export_dir_uncompressed), })}
-        )
+        context
     )
-    assert len(result.tree.geometries) == len(tile_ids)
-    assert len(result.paths_array) == len(tile_ids)
+    assert len(result.tree.geometries) == len(TILE_IDS)
+    assert len(result.paths_array) == len(TILE_IDS)
     result_tile_ids = tuple(sorted(result.export_results.keys()))
-    assert result_tile_ids == tile_ids
+    assert result_tile_ids == TILE_IDS
 
 
-def test_party_walls(resource_db_connection, resource_container,
-                     export_dir_uncompressed, output_data_dir):
+@pytest.mark.slow
+def test_party_walls(context, output_data_dir):
     """Can we compute the party walls and other statistics?"""
-    result = materialize_to_memory(
-        [distribution_tiles_files_index, party_walls_nl],
-        partition_key='10/564/624',
-        resources={"db_connection": resource_db_connection,
-                   "container": resource_container}
-    )
-    df = result.asset_value("party_walls_nl")
-    df.to_csv(str(output_data_dir / "party_walls_nl.csv"))
-    assert result.success
+
+    result = party_walls_nl(context, mock_distribution_tiles_files_index(output_data_dir))
+    assert not result.empty
 
 
-def test_features_file_index(crop_reconstruct_data_dir):
+def test_features_file_index_generator(crop_reconstruct_data_dir):
```

Comment:
I would remove this, because features_file_index is basically the same, but wrapped in an asset.
Plus, test_features_file_index is updating the pickle which is then used by other tests.

Reply:
Yes that is a mistake, I removed the pickling (i only used it to create the test data but forgot to remove it!)